File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
function solution(n, m) {
    function euclide(p, q) {
        if (p % q === 0) return q;
        
        return euclide(q, p % q);
    }
    
    const gcd = euclide(n, m);
    const lcm = n * m / gcd;
    
    return [gcd, lcm];
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def solution(n, m):
    gcd = 1

    for i in range(n, 1, -1):
        if n % i == 0 and m % i == 0:
            gcd = i
            break     
    
    lcm = n * m / gcd
    
    return [gcd, lcm]